hibike/devices/LineFollower/line_follower.cpp

```diff
@@ -32,20 +32,21 @@ uint32_t device_write(uint8_t param, uint8_t* data, size_t len) {
 }
 
 
+
 // You must implement this function.
-// It is called when the device receives a Device Data Update packet.
+// It is called when the device receives a Device Data packet.
 // Modifies data_update_buf to contain the parameter value.
 //    param           -   Parameter index
-//    data_update_buf -   buffer to return data in
-//    buf_len         -   Maximum length of the buffer? TODO: Clarify
+//    data            -   buffer to return data in, little-endian
+//    len             -   length of the buffer left to write to
 //
-//    return          -   sizeof(param) on success; 0 otherwise
+//    return          -   sizeof(param) on success; 0 otherwise 
 
 uint8_t device_read(uint8_t param, uint8_t* data, size_t len) {
   if (MAX_PAYLOAD_SIZE - len < sizeof(uint8_t)|| param > 2 || param < 0) {
     return 0;
   }
   float *data_float = (float *)data;
-  data_float[0] = analogRead(pins[param]);
+  data_float[0] = analogRead(pins[param])/1023;
```

Comment:
Please test this specific line.

It looks like this code is dividing an analog read (an int) by an int (1023).  I think* that this will always be zero, unless analog Read is 1023, when this will return 1.

I believe that the right line would be,
data_float[0] = float(analogRead(pins[param]))/1023;

Which would be a float divided by an int, which is a float.